src/main/java/brave/opentracing/BraveActiveSpan.java

```diff
@@ -0,0 +1,199 @@
+     * @param wrapped the wrapped BraveSpan to which we will delegate all span operations
+     * @param refCount the total number of Continuations of this ActiveSpan (new instances should pass 1)
+     * @param original whether this ActiveSpan instance was created when the span was started (if false, deactivate is a no-op)
+     */
+    BraveActiveSpan(SpanInScope scope, BraveSpan wrapped, AtomicInteger refCount, boolean original) {
+        this.scope = scope;
+        this.wrapped = wrapped;
+        this.refCount = refCount;
+        this.original = original;
+    }
+
+    @Override
+    public void deactivate() {
+        if (original) {
+            if (0 == refCount.decrementAndGet()) {
+                wrapped.finish();
+                scope.close();
+            }
+        }
+    }
+
+    @Override
+    public BraveActiveSpan.Continuation capture() {
+        return new BraveActiveSpan.Continuation();
+    }
+
+    @Override
+    public void close() {
+        deactivate();
+    }
+
+    public Span unwrap() {
+        return wrapped.unwrap();
+    }
+
+    @Override
+    public SpanContext context() {
+        return wrapped.context();
+    }
+
+    @Override
+    public BraveActiveSpan setTag(String key, String value) {
+        wrapped.setTag(key, value);
+        return this;
+    }
+
+    @Override
+    public BraveActiveSpan setTag(String key, boolean value) {
+        wrapped.setTag(key, value);
+        return this;
+    }
+
+    @Override
+    public BraveActiveSpan setTag(String key, Number value) {
+        wrapped.setTag(key, value);
+        return this;
+    }
+
+    @Override
+    public BraveActiveSpan log(Map<String, ?> fields) {
+        wrapped.log(fields);
+        return this;
+    }
+
+    @Override
+    public BraveActiveSpan log(long timestampMicroseconds, Map<String, ?> fields) {
+        wrapped.log(timestampMicroseconds, fields);
+        return this;
+    }
+
+    public static String toAnnotation(Map<String, ?> fields) {
+        return BraveSpan.toAnnotation(fields);
+    }
+
+    public static String joinOnEqualsSpace(Map<String, ?> fields) {
+        return BraveSpan.joinOnEqualsSpace(fields);
+    }
+
+    @Override
+    public BraveActiveSpan log(String event) {
+        wrapped.log(event);
+        return this;
+    }
+
+    @Override
+    public BraveActiveSpan log(long timestampMicroseconds, String event) {
+        wrapped.log(timestampMicroseconds, event);
+        return this;
+    }
+
+    @Override
+    public BraveActiveSpan log(String eventName, Object ignored) {
+        wrapped.log(eventName, ignored);
+        return this;
+    }
+
+    @Override
+    public BraveActiveSpan log(long timestampMicroseconds, String eventName, Object ignored) {
+        wrapped.log(timestampMicroseconds, eventName, ignored);
+        return this;
+    }
+
+    @Override
+    public BraveActiveSpan setBaggageItem(String key, String value) {
+        wrapped.setBaggageItem(key, value);
```

Comment:
is it possible to make a (package private) BraveBaseSpan (vs copy/pasting dispatch stuff which is easy to accidentally get wrong)